http://bugs.debian.org/465197

From: Nikos <nmav@crystal.(none)>
Date: Sun, 17 Feb 2008 18:37:24 +0000 (+0200)
Subject: corrected string handling in parse_general_name. Thanks to Andreas Metzler for pointi ...
X-Git-Url: http://git.savannah.gnu.org/gitweb/?p=gnutls.git;a=commitdiff_plain;h=38613e1dd06563e40c1996d01df88d3f2cbd57a3

corrected string handling in parse_general_name. Thanks to Andreas Metzler for pointing out.
---

diff --git a/lib/x509/x509.c b/lib/x509/x509.c
index a5857b0..93e81bb 100644
--- a/lib/x509/x509.c
+++ b/lib/x509/x509.c
@@ -1043,8 +1043,10 @@ parse_general_name (ASN1_TYPE src, const char *src_name,
       result = asn1_read_value (src, nptr, name, &len);
       *name_size = len;
 
-      if (result == ASN1_MEM_ERROR)
+      if (result == ASN1_MEM_ERROR) {
+          if (is_type_printable(type)) (*name_size)++;
   	  return GNUTLS_E_SHORT_MEMORY_BUFFER;
+      }
 
       if (result != ASN1_SUCCESS)
 	{
